akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala

```diff
@@ -639,18 +663,18 @@ private[akka] class RemoteActorRef private[akka] (
   def isWatchIntercepted(watchee: ActorRef, watcher: ActorRef) = {
     // If watchee != this then watcher should == this. This is a reverse watch, and it is not intercepted
     // If watchee == this, only the watches from remoteWatcher are sent on the wire, on behalf of other watchers
-    watcher != provider.remoteWatcher && watchee == this
+    provider.remoteWatcher.exists(rm => watcher != rm) && watchee == this
   }
 
   def sendSystemMessage(message: SystemMessage): Unit =
     try {
       //send to remote, unless watch message is intercepted by the remoteWatcher
       message match {
         case Watch(watchee, watcher) if isWatchIntercepted(watchee, watcher) =>
-          provider.remoteWatcher ! RemoteWatcher.WatchRemote(watchee, watcher)
+          provider.remoteWatcher.foreach(_ ! RemoteWatcher.WatchRemote(watchee, watcher))
         //Unwatch has a different signature, need to pattern match arguments against InternalActorRef
         case Unwatch(watchee: InternalActorRef, watcher: InternalActorRef) if isWatchIntercepted(watchee, watcher) =>
-          provider.remoteWatcher ! RemoteWatcher.UnwatchRemote(watchee, watcher)
+          provider.remoteWatcher.foreach(_ ! RemoteWatcher.UnwatchRemote(watchee, watcher))
         case _ => remote.send(message, OptionVal.None, this)
```

Comment:
`isWatchIntercepted` will return false and then Watch/Unwatch message is sent anyway here.

We should probably log warning and discard the message instead.